### PR TITLE
Bug fix/6746 run set resetinitial variablevalue & timer value issue

### PR DESCRIPTION
**Pull request by JinendraGhodke**

### CODE REVIEW

File: Ginger/GingerCoreNET/Run/GingerRunner.cs [hunk 1]
```diff
@@ -3719,9 +3719,15 @@ public void ResetRunnerExecutionDetails(bool doNotResetBusFlows=false)
             PublishToALMConfig = null;
             if (doNotResetBusFlows == false)
             {
-                foreach (BusinessFlow bf in BusinessFlows)
+                foreach (BusinessFlow businessFlow in BusinessFlows)
                 {
-                    bf.Reset();
+                    businessFlow.Reset();
+
+                    foreach (VariableBase inputVariable in businessFlow.GetBFandActivitiesVariabeles(true, true))
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/GingerRunner.cs]: why not to include it inside bf.Reset() function?

Comment (JinendraGhodke) [reply Ginger/GingerCoreNET/Run/GingerRunner.cs]: bf.Reset() is called from many places including from Automate tab. 
But i want to reset variable value on run set start.

**Comment by MeniKadosh1** [comment]
So maybe include it in Reset function but do it only based on input bool flag.


Regards,

Meni Kadosh
Automation Tools Development Team Lead
Amdocs Quality Engineering Services
BEAT R&D Center
+972-8-62-40961 (office)

[amdocs-2017-brand-mark-rgb]
Follow us on Facebook<http://www.facebook.com/amdocs/>, Twitter<http://twitter.com/amdocs>, LinkedIn<http://www.linkedin.com/company/amdocs>, YouTube<http://www.youtube.com/amdocsinc>, Google+<https://plus.google.com/105657940751678445194> and the Amdocs blog network<http://blogs.amdocs.com/>.

**Comment by JinendraGhodke** [comment]
I am resetting only the variables for which Set as input value is true.
So I don’t want to put it inside bf.Reset() with the flag, because someone in future might just call this function with this flag as true thinking all variables are reset. Which is confusing.
May be I will add function inside Business flow for ResetInputVariables.  And call this function from separately.
WDYT?


Regards,
Jinendra

**Comment by MeniKadosh1** [comment]
Why reset only for input variables?
I still think we should merge it but your call.


Regards,

Meni Kadosh
Automation Tools Development Team Lead
Amdocs Quality Engineering Services
BEAT R&D Center
+972-8-62-40961 (office)

[amdocs-2017-brand-mark-rgb]
Follow us on Facebook<http://www.facebook.com/amdocs/>, Twitter<http://twitter.com/amdocs>, LinkedIn<http://www.linkedin.com/company/amdocs>, YouTube<http://www.youtube.com/amdocsinc>, Google+<https://plus.google.com/105657940751678445194> and the Amdocs blog network<http://blogs.amdocs.com/>.

**Pull request merged**